active-record/src/main/java/com/iluwatar/active/record/MagicWand.java

```diff
@@ -0,0 +1,264 @@
+            ps.setString(4, this.getCore().name());
+
+            ps.execute();
+
+            ps.close();
+            connection.close();
+            return id;
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void delete() {
+        validateToDelete();
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(DELETE_SQL);
+
+
+            ps.setLong(1, this.getId());
+            ps.execute();
+            ps.close();
+            connection.close();
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void update() {
+        validateToUpdate();
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(UPDATE_SQL);
+
+
+            ps.setDouble(1, this.getLengthInches());
+            ps.setString(2, this.getWood().name());
+            ps.setString(3, this.getCore().name());
+            ps.setLong(4, this.getId());
+
+            ps.execute();
+            ps.close();
+            connection.close();
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static MagicWand find(long id) {
+
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(SELECT_SQL);
+
+            ResultSet rs;
+            ps.setLong(1, id);
+            rs = ps.executeQuery();
+            if (rs.next()) {
+                final MagicWand wand = new MagicWand();
+
+                wand.id = rs.getLong("id");
+                wand.setCore(WandCoreType.valueOf(rs.getString("core")));
+                wand.setWood(WandWoodType.valueOf(rs.getString("wood")));
+                wand.setLengthInches(rs.getDouble("length_inches"));
+
+                return wand;
+            }
+            else {
+                return null;
+            }
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    private void validateToSave() {
+        validateFieldsToBeNotNull();
+        if (this.getId() != null)
+            throw new IllegalStateException("Can not save wand that was previously saved. Use 'update' metod instead.");
+    }
+
+    private void validateToUpdate() {
+        validateFieldsToBeNotNull();
+        if (this.getId() == null) throw new IllegalStateException("Can not update a record without ID specified");
+    }
+
+    private void validateToDelete() {
+        if (this.getId() == null) throw new IllegalStateException("Can not delete a record without ID specified");
+    }
+
+    private void validateFieldsToBeNotNull() {
+        if (this.getLengthInches() == null)
```

Comment:
I am not a big fan of moving all the mandatory fields to constructor as then we end up supporting something like SomeObj(String, String, Int, String, Double, Enum). In addition to this we will mix up the different logic. Probably it's okay to create a wand without a core but it's not okay to save it to some DB due to some business rules. Because of that I used contextual validation - http://martinfowler.com/bliki/ContextualValidation.html.